lib/backend/workspace.dart

```diff
@@ -53,11 +54,12 @@ class WorkspaceController with ChangeNotifier {
     _gridState = const GridViewState(size: 96);
   }
 
-  Future<void> getInfoForDir(Directory dir) async {
-    await _fetcher?.cancel();
+  Future<void> getInfoForDir(fs.File dir) async {
+    /* await */
+    _fetcher?.cancel();
```

Comment:
we'd probably need to wait until the fetcher is done cancelling, then destroying it to free native resources and then afterwards create a new fetcher